bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/SingleThreadModelController.java

```diff
@@ -0,0 +1,225 @@
+/********************************************************************************
+ * Copyright (c) 2021 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0, or the MIT License which is
+ * available at https://opensource.org/licenses/MIT.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR MIT
+ ********************************************************************************/
+package org.eclipse.emfcloud.modelserver.emf.common;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.log4j.Logger;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+
+import io.javalin.http.Context;
+
+/**
+ * A {@link ModelController} that executes all requests in the same thread, to
+ * ensure we use a consistent resource set state.
+ */
+public class SingleThreadModelController implements ModelController {
+
+   private static final Logger LOG = Logger.getLogger(SingleThreadModelController.class);
+
+   private static final AtomicInteger COUNT = new AtomicInteger(0);
+
+   protected final ModelController delegate;
+
+   protected final Thread thread;
+
+   protected final BlockingQueue<Runnable> actionsQueue = new ArrayBlockingQueue<>(100, true);
+
+   protected final String name;
+
+   @Inject
+   public SingleThreadModelController(final @Named("ModelControllerDelegate") ModelController delegate) {
```

Comment:
We should use a constant for the `"ModelControllerDelegate"`, e.g. as for the preferred format annotation in https://github.com/eclipse-emfcloud/emfcloud-modelserver/blob/873c12dbfca2423fbd5a0d3e6fdcf10fb78a58b2/bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/codecs/CodecsManager.java#L32
https://github.com/eclipse-emfcloud/emfcloud-modelserver/blob/873c12dbfca2423fbd5a0d3e6fdcf10fb78a58b2/bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/codecs/DICodecsManager.java#L38

Reply:
Good point. Done :+1: